src/events_watcher/proposal_signing_backend/dkg.rs

```diff
@@ -37,19 +39,25 @@ where
     }
 }
 
+//AnchorUpdateProposal for evm
 #[async_trait::async_trait]
-impl super::ProposalSigningBackend<AnchorUpdateProposal>
+impl<P> super::ProposalSigningBackend<P>
     for DkgProposalSigningBackend<DkgRuntimeApi, DkgConfig>
+where
+    P: Proposal + Sync + 'static + Send,
 {
-    async fn can_handle_proposal(
-        &self,
-        proposal: &AnchorUpdateProposal,
-    ) -> anyhow::Result<bool> {
+    async fn can_handle_proposal(&self, proposal: &P) -> anyhow::Result<bool> {
         let header = proposal.header();
         let resource_id = header.resource_id();
         let storage_api = self.api.storage().dkg_proposals();
-        let src_chain_id =
-            webb_proposals_typed_chain_converter(proposal.src_chain());
+        let proposal_bytes = proposal.to_vec();
+        // get src chain id from bytes
+        let mut src_chain_bytes = [0u8; TYPE_CHAIN_LENGTH];
+        let j = PROPOSAL_LENGTH + TYPE_CHAIN_LENGTH;
+        src_chain_bytes[..TYPE_CHAIN_LENGTH]
+            .copy_from_slice(&proposal_bytes[PROPOSAL_LENGTH..j]);
```

Comment:
This could be like the following, for easier reading.
```suggestion
        src_chain_bytes.copy_from_slice(&proposal_bytes[PROPOSAL_LENGTH..j]);
```

Reply:
Also, This assumes that the current proposal is `AnchorUpdateProposal`, which is not always the case and will panic if used with other proposals.